fitbenchmarking/core/tests/test_fitting_benchmarking.py

```diff
@@ -0,0 +1,824 @@
+"""
+Tests for fitbenchmarking.core.fitting_benchmarking.Fit
+"""
+import os
+import unittest
+from unittest.mock import patch
+import json
+import numpy as np
+
+from fitbenchmarking.controllers.scipy_controller import ScipyController
+from fitbenchmarking.jacobian.analytic_jacobian import Analytic
+from fitbenchmarking.core.fitting_benchmarking import Fit
+from fitbenchmarking.cost_func.weighted_nlls_cost_func import (
+    WeightedNLLSCostFunc)
+from fitbenchmarking.cost_func.nlls_cost_func import NLLSCostFunc
+from fitbenchmarking.parsing.parser_factory import parse_problem_file
+from fitbenchmarking.utils.checkpoint import Checkpoint
+from fitbenchmarking.utils.options import Options
+from fitbenchmarking.utils.fitbm_result import FittingResult
+from fitbenchmarking.utils.exceptions import (FitBenchmarkException,
+                                              IncompatibleCostFunctionError,
+                                              UnsupportedMinimizerError,
+                                              UnknownMinimizerError,
+                                              IncompatibleMinimizerError,
+                                              NoJacobianError,
+                                              NoHessianError,
+                                              ValidationException)
+
+FITTING_DIR = "fitbenchmarking.core.fitting_benchmarking"
+ROOT = os.getcwd()
+DATA_DIR = ROOT + \
+    "/fitbenchmarking/benchmark_problems/NIST/average_difficulty/"
```

Comment:
I am little apprehensive about converting the `benchmark_problems` folder into a python module just to import it in this way. That folder contains no python code and is only being imported by this test file. I think it might be better to get the files simply this way.

`DATA_DIR = os.path.join(os. getcwd(),
                        'fitbenchmarking',
                        'benchmark_problems',
                        'NIST',
                        'average_difficulty')`

Reply:
I think the main reason for doing it that way (i.e. by adding the init file) was to check whether os.getcwd() was the reason for the Windows test error. If you've found out that the error is due to something else, then I think the way you're suggesting here is fine too

Reply:
Quick one on the difference between `os.getcwd()` and the `import benchmark_problems` implementations is that the latter does not depend on where the script is run:
`cd fitbenchmarking/core; pytest .` would only work with the import option.

You could use a relative path from `__FILE__` (I think that's the right var) which should be a bit more robust than `getcwd` if you didn't want to make it a python module.